src/Avalonia.Base/AvaloniaProperty.cs

```diff
@@ -279,14 +273,14 @@ public abstract class AvaloniaProperty : IEquatable<AvaloniaProperty>, IProperty
         /// <returns>A <see cref="AvaloniaProperty{TValue}"/></returns>
         public static AttachedProperty<TValue> RegisterAttached<TOwner, THost, TValue>(
             string name,
-            TValue defaultValue = default(TValue),
+            TValue defaultValue = default!,
```

Comment:
This, and the other `default!` expressions in this file isn't great. Problem is that `default` is an invalid value if `TValue` is not nullable and there's not really any way to fix this without breaking changes.

For 11.0 we probably need to supply separate `class` and `struct` property registration overloads and only provide a default value for the `struct` overloads, but this would be a (pretty big) breaking change.